src/mint/library.cairo

```diff
@@ -630,27 +585,37 @@ namespace CarbonableMinter {
             assert_le(value, max_value_per_tx);
         }
 
-        // [Check] Enough NFTs available
-        let (carbonable_project_address) = CarbonableMinter_carbonable_project_address_.read();
-        let (carbonable_project_slot) = CarbonableMinter_carbonable_project_slot_.read();
-        let (total_value) = IERC3525.totalValue(
-            carbonable_project_address, carbonable_project_slot
-        );
+        // [Effect] If remaining value is lower than specified value and force is enabled
+        // Then replace the specified value by the remaining value otherwize keep the value unchanged
         let (max_value) = CarbonableMinter_max_value_.read();
         let (reserved_value) = CarbonableMinter_reserved_value_.read();
-        let (value_uint256) = _felt_to_uint(value);
-        let (value_after_buy) = SafeUint256.add(total_value, value_uint256);
-        let (available_value) = _felt_to_uint(max_value - reserved_value);
-        let (enough_available) = uint256_le(value_after_buy, available_value);
+        let (available_value_u256) = _felt_to_uint(max_value - reserved_value);
+
+        let (project_address) = CarbonableMinter_carbonable_project_address_.read();
+        let (project_slot) = CarbonableMinter_carbonable_project_slot_.read();
+        let (total_value) = IERC3525.totalValue(project_address, project_slot);
+
+        let (remaining_value) = SafeUint256.sub_le(available_value_u256, total_value);
+        let (value_u256) = _felt_to_uint(value);
+        let (is_lower) = uint256_lt(remaining_value, value_u256);
+
+        // Set value = condition * remaining_value + (1 - condition) * value
+        let (condition) = _felt_to_uint(is_lower * force);
```

Comment:
abuséd
Uint256 really is annoying here.
I think you can be more efficinet here by computing the multiplication directly as Uint256(cond*low, cond*high)